commands/command_clone.go

```diff
@@ -79,8 +81,36 @@ func cloneCommand(cmd *cobra.Command, args []string) {
 		fetchRef("HEAD", include, exclude)
 	} else {
 		pull(include, exclude)
+
+		err := postCloneSubmodules(args)
+		if err != nil {
+			Exit("Error performing 'git lfs pull' for submodules: %v", err)
+		}
+	}
+
+}
+
+func postCloneSubmodules(args []string) error {
+	// In git 2.9+ the filter option will have been passed through to submodules
+	// So we need to lfs pull inside each, if the original clone command
+	if !git.Config.IsGitVersionAtLeast("2.9.0") {
+		// In earlier versions submodules would have used smudge filter
+		return nil
+	}
+	// Also we only do this if --recursive or --recurse-submodules was provided
+	if !cloneFlags.Recursive && !cloneFlags.RecurseSubmodules {
+		return nil
 	}
 
+	// Use `git submodule foreach --recursive` to cascade into nested submodules
+	// Also good to call a new instance of git-lfs rather than do things
+	// inside this instance, since that way we get a clean env in that subrepo
+	cmd := subprocess.ExecCommand("git", "submodule", "foreach", "--recursive",
```

Comment:
As discussed in #1172 the reason for using `git submodule foreach` is that you get recursion (submodules within submodules) for free - re-implementing the recursion ourselves is a complication I wanted to avoid (your code template above won't handle it and would actually fail the integration test since it specifically tests for this). 

Also since transfers in lfs pull are already parallelised I don't think we necessarily gain very much by also parallelising across submodules. The main bottleneck is the transfers so I don't think the added benefit of parallel submodule parsing is worth the extra complexity.